### PR TITLE
[MLIR][Flang][OpenMP] Implement lowering simd aligned to MLIR

### PR DESCRIPTION
**Pull request by harishch4**
Rebased @DominikAdamski patch: https://reviews.llvm.org/D142722

### CODE REVIEW

File: flang/lib/Lower/OpenMP/ClauseProcessor.cpp
Comment (tblah) [inline flang/lib/Lower/OpenMP/ClauseProcessor.cpp:1055]: nit: instead of casting `objects.size()`, it would be better to define `i` with the same type as `objects.size()`. Also please use C++ style casts.

File: flang/test/Lower/OpenMP/simd_aarch64.f90
Comment (tblah) [inline flang/test/Lower/OpenMP/simd_aarch64.f90:2]:
```suggestion
! The default alignment for AARCH64 is 0 so we do not emit aligned clause
```

**Review by tblah — COMMENTED** [review]
Thanks! Just minor comments

**Review by tblah — APPROVED** [review]
Thank you for addressing my comments. LGTM

File: flang/lib/Lower/OpenMP/ClauseProcessor.cpp
Comment (skatrak) [inline flang/lib/Lower/OpenMP/ClauseProcessor.cpp:1008]: Wouldn't it help to call `features.getFeatures()` here instead and iterate through the list of `StringAttr` directly, rather than producing the comma-separated string to then split it again?

File: flang/lib/Lower/OpenMP/ClauseProcessor.h
Comment (skatrak) [inline flang/lib/Lower/OpenMP/ClauseProcessor.h:141]: Nit: The method list should be alphabetically sorted. In this case, right before `processAllocate`.

File: flang/lib/Lower/OpenMP/OpenMP.cpp
Comment (skatrak) [inline flang/lib/Lower/OpenMP/OpenMP.cpp:1065]: Nit: Sort alphabetically here as well (helps with consistency in the output MLIR) unless it needs to go after processing some other specific clause, in which case it should be documented.

**Review by skatrak — APPROVED** [review]
Thank you for this work! I just have some minor comments, but looks good. No need to wait for another approval from me after making these small changes.

File: flang/lib/Lower/OpenMP/ClauseProcessor.cpp
Comment (harishch4) [reply flang/lib/Lower/OpenMP/ClauseProcessor.cpp]: Yeah, that seems cleaner. Updated code.

**Pull request merged**
